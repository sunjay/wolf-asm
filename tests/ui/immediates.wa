section .code

add $1, $2
add $1, 100
add $1, 0x1fffffffffff
# invalid: more than 46-bits
add $1, 0x7fffffffffff
add $1, 0x1000000000000

cmp $2, $1
cmp 100, $1
cmp 0x1fffffffffff, $1
# invalid: more than 46-bits
cmp 0x7fffffffffff, $1
cmp 0x1000000000000, $1

store8 -4($2), $3
store8 -8($2), 32
store8 -8($2), 0x1fffffff
store8 0x1ffffff, 0x1ffffff
# invalid: more than 30-bits
store8 -8($2), 0x3fffffff
# invalid: more than 26-bits
store8 0x7fffffff, 0x7fffffff

divr $3, $2, 0x7fffffffff
# invalid: more than 40-bits
divr $3, $2, 0x1ffffffffff

push 0x7ffffffffffff
# invalid: more than 52-bits
push 0x1fffffffffffff
