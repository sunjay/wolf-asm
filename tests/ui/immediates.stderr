[../tests/ui/immediates.wa:7:9-7:22] error: immediate value `140737488355327` (`0x7fffffffffff`) for this instruction must fit in a 46-bit signed number
[../tests/ui/immediates.wa:7:9-7:22] note: that means the value must be between `-35184372088832` and `35184372088831` (`0x1fffffffffff`)

[../tests/ui/immediates.wa:8:9-8:23] error: immediate value `281474976710656` (`0x1000000000000`) for this instruction must fit in a 46-bit signed number
[../tests/ui/immediates.wa:8:9-8:23] note: that means the value must be between `-35184372088832` and `35184372088831` (`0x1fffffffffff`)

[../tests/ui/immediates.wa:14:5-14:18] error: immediate value `140737488355327` (`0x7fffffffffff`) for this instruction must fit in a 46-bit signed number
[../tests/ui/immediates.wa:14:5-14:18] note: that means the value must be between `-35184372088832` and `35184372088831` (`0x1fffffffffff`)

[../tests/ui/immediates.wa:15:5-15:19] error: immediate value `281474976710656` (`0x1000000000000`) for this instruction must fit in a 46-bit signed number
[../tests/ui/immediates.wa:15:5-15:19] note: that means the value must be between `-35184372088832` and `35184372088831` (`0x1fffffffffff`)

[../tests/ui/immediates.wa:22:16-22:25] error: immediate value `1073741823` (`0x3fffffff`) for this instruction must fit in a 30-bit signed number
[../tests/ui/immediates.wa:22:16-22:25] note: that means the value must be between `-536870912` and `536870911` (`0x1fffffff`)

[../tests/ui/immediates.wa:24:8-24:17] error: immediate value `2147483647` (`0x7fffffff`) for this instruction must fit in a 26-bit signed number
[../tests/ui/immediates.wa:24:8-24:17] note: that means the value must be between `-33554432` and `33554431` (`0x1ffffff`)

[../tests/ui/immediates.wa:24:20-24:29] error: immediate value `2147483647` (`0x7fffffff`) for this instruction must fit in a 26-bit signed number
[../tests/ui/immediates.wa:24:20-24:29] note: that means the value must be between `-33554432` and `33554431` (`0x1ffffff`)

[../tests/ui/immediates.wa:28:14-28:26] error: immediate value `2199023255551` (`0x1ffffffffff`) for this instruction must fit in a 40-bit signed number
[../tests/ui/immediates.wa:28:14-28:26] note: that means the value must be between `-549755813888` and `549755813887` (`0x7fffffffff`)

[../tests/ui/immediates.wa:32:6-32:21] error: immediate value `9007199254740991` (`0x1fffffffffffff`) for this instruction must fit in a 52-bit signed number
[../tests/ui/immediates.wa:32:6-32:21] note: that means the value must be between `-2251799813685248` and `2251799813685247` (`0x7ffffffffffff`)

error: aborting due to 9 previous errors

